lab3/README.md

```diff
@@ -0,0 +1,144 @@
+# Lab 3: Deploying free5GC
+
+## Introduction
+
+In Lab3, you will learn how to deploy free5GC with docker and set up interface network.
+
+## Goals of this lab
+
+- Understand listening address 
+- Learn how to deploy free5GC with docker 
+- Learn how to use free5GC
+
+## Preparation
+
+* Install GTP5G: refer [GTP5G github](https://github.com/free5gc/gtp5g) to install
+
+* Install Docker Engine and Docker Compose: refer [Docker Website](https://docs.docker.com/engine/install/ubuntu/) to install 
+
+## Listening Address
+
+Listening Address the IP address and port used by a server to listen for connections from clients. These addresses and ports are used to accept requests from clients. ex: 192.168.100.101:12345
+
+In free5GC, each NF (Network Function) has its own listening addresses used to receive and process requests sent by other NFs.
+
+## N2 & N3 & N4 interface
+![architecture](./images/architecture.png)
+* N2: handles control signaling between the `gNB` and `AMF`
+* N3: manages user data transmission between the `gNB` and `UPF`
+* N4: controls user plane configuration and session management between the `SMF` and `UPF`
+
+These three interfaces are the most important interfaces in the 5G system. Therefore, this lab will teach you how to configure the addresses for these interfaces.
+
+### N6 NAT
+Interface to data network.
+
+UPF will performs NAT on packets output through the N6 interface. Rules are set up in `upf-iptables.sh`.
+
+### N9
+Interface for two UPFs communication.
+
+## Exercise: Configure N2 & N3 & N4 interface in Docker Compose
+In this exercise, we will use docker bridge network to set up these three interfaces.
+
+About docker network, you can refer [Docker Networking – Basics, Network Types & Examples](https://spacelift.io/blog/docker-networking) to learn more information.
+
+In bottom of deploy_exercise.yaml, you can find network setting.
+```yaml
+networks:
+  privnet:
+    ipam:
+      driver: default
+      config:
+        - subnet: 10.100.200.0/24
+    driver_opts:
+      com.docker.network.bridge.name: br-free5gc
+```
+For example, privnet is the bridge network for NFs internal communicaion. ex: Nnrf, Nudm...
+
```

Comment:
Explain how the bridge network runs, and give an example (e.g. how to use brctl).